crates/dojo-core/src/auth/systems.cairo

```diff
@@ -7,100 +7,157 @@ struct Route {
-        IWorldDispatcher { contract_address: world_address }.set_entity(
-            dojo_core::string::ShortStringTrait::new('Status'), (r.role_id, r.resource_id).into(), 0_u8, calldata.span());
+    fn execute(route: Route) {
+        // Set scoped role
+        commands::set_entity(
+            (route.target_id, route.resource_id).into(), (AuthRole { id: route.role_id })
+        );
 
-        _set_authorization_routing(ref routing, world_address);
+        // Set status
+        commands::set_entity(
+            (route.role_id, route.resource_id).into(),
+            (AuthStatus { is_authorized: bool::True(()) })
+        );
     }
 }
 
 #[system]
-mod Authorize {
+mod IsAccountAdmin {
     use traits::Into;
+    use starknet::get_tx_info;
+    use box::BoxTrait;
     use dojo_core::integer::u250;
-    use dojo_core::auth::components::Status;
-    use dojo_core::auth::components::Role;
+    use dojo_core::auth::components::AuthStatus;
+    use dojo_core::auth::components::AuthRole;
+
+    fn execute() -> bool {
+        // Get calling account contract address
+        let caller = get_tx_info().unbox().account_contract_address; // tx origin
+        let role = commands::<AuthRole>::entity(caller.into());
+        // Authorize if role is Admin
+        role.id.into() == 'Admin'
+    }
+}
 
-    fn execute(caller_id: u250, resource_id: u250) {
-        let role = commands::<Role>::entity(caller_id.into());
-        let authorization_status = commands::<Status>::entity(
-            (role.id, resource_id).into()
+#[system]
+mod IsAuthorized {
+    use traits::Into;
+    use dojo_core::integer::u250;
+    use dojo_core::auth::components::AuthStatus;
+    use dojo_core::auth::components::AuthRole;
+
+    fn execute(target_id: u250, resource_id: u250) -> bool {
+        // Get World level role
+        let maybe_role = commands::<AuthRole>::try_entity(target_id.into());
+        let role = match maybe_role {
+            Option::Some(role) => role.id.into(),
+            Option::None(()) => 0,
+        };
+
+        // Get component-scoped role
+        let maybe_scoped_role = commands::<AuthRole>::try_entity((target_id, resource_id).into());
+        let scoped_role = match maybe_scoped_role {
+            Option::Some(scoped_role) => scoped_role.id.into(),
+            Option::None(_) => 0,
+        };
+
+        // Get authorization status for scoped role
+        let maybe_authorization_status = commands::<AuthStatus>::try_entity(
+            (scoped_role, resource_id).into()
         );
-        assert(authorization_status.is_authorized, 'not authorized');
+        let authorization_status = match maybe_authorization_status {
+            Option::Some(authorization_status) => authorization_status.is_authorized,
+            Option::None(_) => bool::False(()),
+        };
+        // Authorize if system's role is Admin or authorization status is true
+        role == 'Admin' | authorization_status
```

Comment:
Modified this one to check if the calling System has an authorized role (Component specific)  or has `Admin` role (Any Component)